### ISSUE TITLE
Make "indent" rule support ObjectPattern

### ISSUE THREAD

**Issue by maxenglander** [eslint/eslint#7670]
**What version are you using?**

```
 $ ./bin/eslint --version
 v3.11.0
```

**What did you do?**

```
$ cat | eslint --env es6  \
>              --parser-options '{"ecmaVersion":7,"sourceType":"module"}' \
>              --rule '{"indent":["error",4],"}' --no-eslintrc --stdin
const {
 a,
b,
   c
} = foo;^D
```

**What happened?**

No errors were reported by eslint.

**What did you expect to happen?**

Would like it if indent rule violations were reported for the `foo` const and for other object patterns.

**Comment by gyandeeps** [comment]
It does report patterns for this. You can use the demo app to try this code.
http://eslint.org/demo/

**Comment by maxenglander** [comment]
Hey @gyandeeps I wrote the example in backwards, sorry. I originally wrote

```javascript
const foo = {
    a,
  b,
      c
};
```

When I meant to write
```javascript
const {
 a,
b,
   c
} = foo;^D
```

**Comment by platinumazure** [comment]
@not-an-aardvark Does your rewrite cover this case?

**Comment by ilyavolodin** [comment]
This is object rest/spread. Since this is experimental feature, we don't really have rules that support it.

**Comment by not-an-aardvark** [comment]
* Yes, the indent rewrite covers this case.
* This is destructuring, not object rest/spread, so I think it's reasonable for the rule to support it.
* I think this is a duplicate of https://github.com/eslint/eslint/issues/6813.

**Comment by maxenglander** [comment]
@ilyavolodin understood. Would you be open to a PR that allows the user to opt-in to for errors for this use case?

@not-an-aardvark I have a simple fix for this ready to go. Should I hold off on submitting a PR since it's already covered by your re-write?

**Comment by gyandeeps** [comment]
closing as duplicate of #6813 ... lets continue the conversation their.
